component-classes/index.js

```diff
@@ -386,7 +386,7 @@ export const input = {
   default: 'block text-m mb-0 leading-m s-text s-bg s-border hover:s-border-hover active:s-border-selected rounded-4 py-12 px-8 block border-1 w-full focusable focus:[--w-outline-offset:-2px] caret-current',
   textArea: 'min-h-[42] sm:min-h-[45]',
   disabled: 's-bg-disabled-subtle s-border-disabled hover:s-border-disabled! s-text-disabled pointer-events-none',
-  invalid: 's-border-negative s-text-negative!',
+  invalid: 's-border-negative s-text-negative! hover:s-border-negative-hover! outline-[--w-s-color-border-negative]!',
```

Comment:
I decided to fix the issue using !important rule for now to not block its release. The proper way to apply colours in different states would be part of a bigger rewrite of input and select classes, which hopefully will start after Easter.